src/frequenz/client/dispatch/_types.py

```diff
@@ -0,0 +1,442 @@
+        )
+
+    @classmethod
+    def from_protobuf(
+        cls, pb_object: PBDispatchCreateRequest
+    ) -> "DispatchCreateRequest":
+        """Convert a protobuf dispatch create request to a dispatch.
+
+        Args:
+            pb_object: The protobuf dispatch create request to convert.
+
+        Returns:
+            The converted dispatch.
+        """
+        return DispatchCreateRequest(
+            microgrid_id=pb_object.microgrid_id,
+            type=pb_object.type,
+            start_time=pb_object.start_time.ToDatetime().replace(tzinfo=timezone.utc),
+            duration=timedelta(seconds=pb_object.duration),
+            selector=_component_selector_from_protobuf(pb_object.selector),
+            is_active=pb_object.is_active,
+            is_dry_run=pb_object.is_dry_run,
+            payload=MessageToDict(pb_object.payload),
+            recurrence=RecurrenceRule.from_protobuf(pb_object.recurrence),
+        )
+
+    def to_protobuf(self) -> PBDispatchCreateRequest:
+        """Convert a dispatch to a protobuf dispatch create request.
+
+        Returns:
+            The converted protobuf dispatch create request.
+        """
+        pb_request = PBDispatchCreateRequest()
+
+        pb_request.microgrid_id = self.microgrid_id
+        pb_request.type = self.type
+        pb_request.start_time.FromDatetime(self.start_time)
+        pb_request.duration = int(self.duration.total_seconds())
+        pb_request.selector.CopyFrom(_component_selector_to_protobuf(self.selector))
+        pb_request.is_active = self.is_active
+        pb_request.is_dry_run = self.is_dry_run
+        pb_request.payload.update(self.payload)
+        pb_request.recurrence.CopyFrom(self.recurrence.to_protobuf())
+
+        return pb_request
+
+
+@dataclass(kw_only=True)
+class Dispatch(DispatchCreateRequest):
+    """Represents a dispatch operation within a microgrid system."""
+
+    id: int
+    """The unique identifier for the dispatch."""
+
+    create_time: datetime
+    """The creation time of the dispatch in UTC. Set when a dispatch is created."""
+
+    update_time: datetime
+    """The last update time of the dispatch in UTC. Set when a dispatch is modified."""
+
+    @classmethod
+    def from_request(
+        cls,
+        _request: DispatchCreateRequest,
+        _id: int,
+        create_time: datetime,
+        update_time: datetime,
+    ) -> "Dispatch":
+        """Initialize a Dispatch object from a request.
+
+        Args:
+            _request: The dispatch create request.
+            _id: The unique identifier for the dispatch.
+            create_time: The creation time of the dispatch in UTC.
+            update_time: The last update time of the dispatch in UTC.
+
+        Returns:
+            The initialized dispatch.
+        """
+        return Dispatch(
+            id=_id,
+            create_time=create_time,
+            update_time=update_time,
+            **_request.__dict__,
+        )
```

Comment:
This is not even used, right?

Reply:
for tests mostly

Reply:
Why not adding it to the `tests/` folder then? I you need to reuse it from different files you can just add something like `tests/utils.py`.